packages/ui/src/main.ts

```diff
@@ -0,0 +1,110 @@
+import show, { type WalletProviderWithStoreVersion } from "./modal"
+import Bowser from "bowser"
+import sn, { type StarknetWindowObject } from "get-starknet-core"
+
+type StoreVersion = "chrome" | "firefox" | "edge"
+
+function getStoreVersionFromBrowser(): StoreVersion {
+  const browserName = Bowser.getParser(window.navigator.userAgent)
+    .getBrowser()
+    .name?.toLowerCase()
+  switch (browserName) {
+    case "firefox":
+      return "firefox"
+    case "microsoft edge":
+      return "edge"
+    default:
+      return "chrome"
+  }
+}
+
+export interface ConnectOptions {
+  modalMode?: "alwaysAsk" | "canAsk" | "neverAsk"
+  modalTheme?: "light" | "dark" | "system"
+  storeVersion?: StoreVersion
+}
+
+export interface DisconnectOptions {
+  clearDefaultWallet?: boolean
+}
+
+export const connect = async ({
+  modalMode = "canAsk",
+  storeVersion = getStoreVersionFromBrowser(),
+  modalTheme,
+}: ConnectOptions = {}): Promise<StarknetWindowObject | null> => {
+  const enableWithVersion = async (wallet: StarknetWindowObject | null) => {
+    if (!wallet) {
+      return null
+    }
+    return sn.enable(wallet, { starknetVersion: "v4" }).catch(() => null)
+  }
+
+  const lastWallet = await sn.getLastConnectedWallet()
+  if (modalMode !== "alwaysAsk" && lastWallet) {
+    return enableWithVersion(lastWallet)
+  }
+
+  const preAuthorizedWallets = await sn.getPreAuthorizedWallets({
+    exclude: [lastWallet?.id],
```

Comment:
not sure I understand this exclusion?

Reply:
the wallets of the prior step will always be in the list of wallets to show already. So we can ignore them for every further step